### ISSUE TITLE
verify-elf: ERROR: ./usr/lib64/liburing.so.2.4: undefined symbol: memset

### ISSUE THREAD

**Issue by vt-alt** [axboe/liburing#874]
We have our build system lint (verify-elf) reporting error for liburing.so binary for v2.4 on `aarch64` (only, on x86_64 there's no problem):

    verify-elf: ERROR: ./usr/lib64/liburing.so.2.4: undefined symbol: memset

```
builder@aarch64:~/RPM/BUILD/liburing-2.4$ ldd ./src/liburing-ffi.so.2.4
        statically linked
builder@aarch64:~/RPM/BUILD/liburing-2.4$ nm -D ./src/liburing-ffi.so.2.4
0000000000000000 A LIBURING_2.4
0000000000006344 T __io_uring_buf_ring_cq_advance@@LIBURING_2.4
...
0000000000004150 T io_uring_wait_cqe_timeout@@LIBURING_2.4
0000000000004080 T io_uring_wait_cqes@@LIBURING_2.4
                 U memset
```

Perhaps, something is linked incorrectly or `memset` is defined incorrectly.

**Comment by ammarfaizi2** [comment]
> Perhaps, something is linked incorrectly or `memset` is defined incorrectly.

Since commit db5403e ("nolibc: Do not define memset() function in liburing"),
liburing no longer has a function called `memset()`. I changed it to
`__uring_memset()` because it conflicted with the `memset()` function
from libc.

I'm not sure what's going on with aarch64 here. Need more clue...

**Comment by axboe** [comment]
Yeah that's funky, I see the same on aarch64 here:

```
axboe@m1max ~/gi/liburing (master)> nm -D src/liburing-ffi.so.2.4 | grep " U " 0.013s
                 U memset
```

**Comment by axboe** [comment]
And botched that for !nolibc... Not batting 1000 this morning, hopefully current git is sane.

**Comment by vt-alt** [comment]
Thanks much! [Btw, you have typo in commit message `s/aarc64/aarch64/`]

**Comment by axboe** [comment]
Yep I will write an announcement, but deferred it since I wanted to make sure we didn't have a silly issue. Which turned out to be a good thing! You don't really get good coverage before it's tagged, which is a bit of a problem.
